### PR TITLE
API filtering and sorting improvements

### PR DESCRIPTION
**Pull request by JakubFrejlach**
This PR adds more granular filtering with the ability to filter on related models as well and adds support for ordering the API response by specified field (ascending or descending) for Flaws, Affects and Trackers.

Closes OSIDB-667, OSIDB-668

### CODE REVIEW

**Comment by costaconrado** [comment]
Nit: OSIDB-668 did not entered on the Jira automation because it only gets if you add one "Closes OSIDB-XXX" per issue in the description.

**Comment by costaconrado** [comment]
Optional: I think changes in automated generated schema, secrets and changelog could be squashed in a single commit for a cleaner git history, but it is just a personal preference.

**Comment by costaconrado** [comment]
Made some optional suggestions but LGTM.

**Review by costaconrado — APPROVED** [review]

File: .secrets.baseline [hunk 1]
```diff
@@ -274,5 +274,5 @@
       }
     ]
   },
-  "generated_at": "2023-03-31T18:16:32Z"
+  "generated_at": "2023-04-04T19:16:10Z"
```
Comment (osoukup) [inline .secrets.baseline:277]: This change seems unnecessary to me when no secrets have changed.

**Review by osoukup — APPROVED** [review]
Added two comments to consider but generally this is a very good improvement! LGTM

File: osidb/filters.py [hunk 2]
```diff
@@ -139,28 +201,162 @@ class Meta:
...
+        }
 
 
 class TrackerFilter(DistinctFilterSet):
+
+    DISTINCT_FIELDS_PREFIXES = ("affects__",)
+
+    embargoed = BooleanFilter(field_name="embargoed")
+    affects__embargoed = BooleanFilter(field_name="affects__embargoed")
+    flaw__embargoed = BooleanFilter(field_name="flaw__embargoed")
```
[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply osidb/filters.py:292]: You are correct, most likely a error from copying the filters around and changing the prefixes. Thanks!

File: .secrets.baseline (hunk 1 above)
Comment (JakubFrejlach) [reply .secrets.baseline]: Yep. I've run update-secrets and saw some changes in git status so just commited without checking that the date changed only.

**Comment by JakubFrejlach** [comment]
> Nit: OSIDB-668 did not entered on the Jira automation because it only gets if you add one "Closes OSIDB-XXX" per issue in the description. (It will not automatically close nor add the PR url in the Jira issue.)

Thx, fixed.

**Comment by JakubFrejlach** [comment]
> Optional: I think changes in automated generated schema, secrets and changelog could be squashed in a single commit for a cleaner git history, but it is just a personal preference.

Good idea. My old habit to keep those separated but it makes more sense to squash them. :smile:

**Pull request merged**